lib/assigner.rb

```diff
@@ -463,6 +452,31 @@ def unassign(silent: false, deactivate: false)
 
   private
 
+  def invite_user(user)
+    return if topic.all_allowed_users.exists?(id: user.id)
+
+    guardian.ensure_can_invite_to!(topic)
+    topic.invite(@assigned_by, user.username)
+  end
+
+  def invite_group(group)
+    return if topic.topic_allowed_groups.exists?(group_id: group.id)
+    if topic
+         .all_allowed_users
+         .joins("RIGHT JOIN group_users ON group_users.user_id = users.id")
+         .where("group_users.group_id = ? AND users.id IS NULL", group.id)
+         .empty?
```

Comment:
```suggestion
         .exists?
```

Small suggestion but I think we can be more efficient here by just doing a `SELECT 1` query

Reply:
I'm still using `SELECT 1` here. The [`empty?`](https://apidock.com/rails/v5.2.3/ActiveRecord/Relation/empty%3F) method is equal to the `!exists?` method.

Reply:
O icic. I thought `empty?` is an array method. All good here :+1: Thank you for clarifying.